### PR TITLE
add BetterDiscord-Store

### PR DESCRIPTION
**Pull request by TheCommieAxolotl**
## Description

An extension to search through the [BetterDiscord](https://betterdiscord.app/) store
(partially based off elm-search extension)

## Screencast

https://user-images.githubusercontent.com/87679354/153703579-4a698ad9-cd3d-4227-811f-769a49c34b5a.mov

If there's an error it will show in a toast

ErrorToast

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/betterdiscord-search/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "betterdiscord-store",
+  "title": "BetterDiscord Store",
+  "description": "Allows you to search and view BetterDiscord Addons inside Raycast",
```
Comment (pernielsentikaer) [inline extensions/betterdiscord-search/package.json:5]:
```suggestion
  "description": "Allows you to search and view BetterDiscord addons inside Raycast",
```

File: extensions/betterdiscord-search/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "icon": "bd1.png",
+  "author": "TheCommieAxolotl",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Find Addons",
+      "subtitle": "BetterDiscord",
+      "description": "Search for a BetterDiscord Addon by Name",
```
Comment (pernielsentikaer) [inline extensions/betterdiscord-search/package.json:14]:
```suggestion
      "description": "Search for a BetterDiscord addon by name",
```

File: extensions/betterdiscord-search/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
+import { Detail, showToast, Toast, Image, Icon, useNavigation, Action, ActionPanel, List } from "@raycast/api";
```
Comment (pernielsentikaer) [inline extensions/betterdiscord-search/src/index.tsx:1]:
```suggestion
import { Detail, showToast, Toast, Icon, useNavigation, Action, ActionPanel, List } from "@raycast/api";
```

File: extensions/betterdiscord-search/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+  return {
+    state: state,
+    search: search,
+  };
+}
+
+export default function Command() {
+  const { push, pop } = useNavigation();
+  const { state, search } = useSearch();
```
Comment (pernielsentikaer) [inline extensions/betterdiscord-search/src/index.tsx:83]:
```suggestion
  const { push } = useNavigation();
  const { state } = useSearch();
```
Unused constants?

File: extensions/betterdiscord-search/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            <Action title="Back" onAction={pop} />
+          </ActionPanel>
+        }
+      />
+    );
+  }
+}
+
+async function performSearch(searchText: string, signal: AbortSignal): Promise<SearchResult[]> {
+  const params = new URLSearchParams();
```
Comment (pernielsentikaer) [inline extensions/betterdiscord-search/src/index.tsx:144]:
```suggestion
  const params = new URLSearchParams();
```
Leftover or forgotten?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @TheCommieAxolotl 👋 

Thanks for your contribution 💪 

I have some feedback ready for you to address:

- Avoid having a list of actions where some have icons and some don't

- An idea could be to add a bit more data to the list, maybe like this:

k3YHEl12b1EAv6Eh

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by TheCommieAxolotl** [comment]
Thanks for the review!

Would this list/action be better?

**Comment by pernielsentikaer** [comment]
Hi @TheCommieAxolotl 👋 

It seems like you have a Prettier code style error, can you please format the code with Prettier?

The icon look very nice, let's use that (and the download part) 😄 💪

File: extensions/betterdiscord-search/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+    return (
+      <Detail
+        navigationTitle="Details"
+        markdown={props.description}
+        actions={
+          <ActionPanel title="Detail">
+            <Action title="Back" onAction={pop} />
+          </ActionPanel>
+        }
```
Comment (pernielsentikaer) [inline extensions/betterdiscord-search/src/index.tsx:139]: I think this should be removed, everywhere is escape used, and we would like to keep it simple and the same – would you mind removing this?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Awesome, it turned out very nice 🔥 

Last correction and we're good to go, very nice @TheCommieAxolotl 💪

**Comment by TheCommieAxolotl** [comment]
Thanks, i've removed the action so it should all be good now 👍🏻

**Review by pernielsentikaer — APPROVED** [review]
Thanks for the update @TheCommieAxolotl 🔥 

I'm going to accept it right away! 💪 

Thanks for your contribution 💯

**Review by mathieudutour — APPROVED** [review]

**Pull request merged**